tests/integration/conftest.py

```diff
@@ -122,13 +122,20 @@ def factory(
         c2r_runtime.logfile_read = sys.stdout
         try:
             yield c2r_runtime
-        except Exception:
-            c2r_runtime.close()
-            raise
-        else:
             c2r_runtime.expect(pexpect.EOF)
             c2r_runtime.close()
+        # Check if child is still alive, if so, send SIGINT
+        # this handles the TIMEOUT exception - if the process is still alive,
+        # the EOF is not raised, the process gets terminated.
+        # If pexpect.EOF exception is not raised (timeouts after 15 minutes)
+        # force terminate the process.
```

Comment:
Please put the comment below the `finally:`. That way it is more clear to what it belongs to

Reply:
done @kokesak 
I have also added a couple of assertions to some tests.